javascript/react/redux-i/understanding-pure-impure-functions.md

```diff
@@ -29,18 +29,18 @@ Example of a simple pure function:
 
 ```javascript
 function doublePure(x) {
-  return 2*x;
- }
+  return 2 * x;
+}
 ```
 
 **Pure** functions must always return a new object and not modify the argument. For example, if we want to decrement every element of an `array`, the **pure** way to do it's by creating a new array with the same elements, but decremented.
 
-On the contrary, **impure** functions don't have a predictable behaviour as they might call the network, the database, modify the **DOM** or even alter the value/s passed to them.
+On the contrary, **impure** functions don't have a predictable behavior as they might call the network, the database, modify the **DOM** or even alter the value/s passed to them.
 
 Example:
 ```javascript
 function doubleImpure(x) {
-  x= 2 * x;
+  x = 2 * x;
```

Comment:
hmmmm this isn't really impure. The changes do not propagate outside `doubleImpure` function.

These are better examples for pure vs impure:

```js
function doublePure(x) {
  // do not change x.
  // instead return a new object
  // with the change applied to it.
  return { ...x, num: x.num * 2 };
}

function doubleImpure(x) {
  // change the same object
  // this mutation affects the object
  // outside the lifetime of doubleImpure
  // which is not "pure"
  x.num = x.num * 2;
  return x;
}
```

Reply:
I've changed the examples to the ones suggested. I've also added the **pure** vs **impure** example in which we show predictable and consistent output for the same input.